src/main/java/uk/gov/hmcts/reform/dg/docassembly/service/CdamService.java

```diff
@@ -0,0 +1,107 @@
+package uk.gov.hmcts.reform.dg.docassembly.service;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.document.am.feign.CaseDocumentClientApi;
+import uk.gov.hmcts.reform.ccd.document.am.model.Classification;
+import uk.gov.hmcts.reform.ccd.document.am.model.Document;
+import uk.gov.hmcts.reform.ccd.document.am.model.DocumentUploadRequest;
+import uk.gov.hmcts.reform.ccd.document.am.model.UploadResponse;
+import uk.gov.hmcts.reform.dg.docassembly.dto.ByteArrayMultipartFile;
+import uk.gov.hmcts.reform.dg.docassembly.dto.CreateTemplateRenditionDto;
+import uk.gov.hmcts.reform.dg.docassembly.service.exception.DocumentTaskProcessingException;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.EnumSet;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.UUID;
+
+@Service
+public class CdamService {
+
+    @Autowired
+    private CaseDocumentClientApi caseDocumentClientApi;
+
+
+    public File downloadFile(String auth, String serviceAuth, UUID documentId) throws
+            IOException, DocumentTaskProcessingException {
+
+        ResponseEntity<Resource> response =  caseDocumentClientApi.getDocumentBinary(auth, serviceAuth, documentId);
+        HttpStatus status = null;
+
+        if (Objects.nonNull(response)) {
+            status = response.getStatusCode();
+            var byteArrayResource = (ByteArrayResource) response.getBody();
+            if (Objects.nonNull(byteArrayResource)) {
+                try (var inputStream = byteArrayResource.getInputStream()) {
+                    var document = caseDocumentClientApi.getMetadataForDocument(auth, serviceAuth, documentId);
+                    var originalDocumentName = document.originalDocumentName;
+                    var fileType = FilenameUtils.getExtension(originalDocumentName);
+                    var fileName = "document." + fileType;
```

Comment:
what happens if there are 2 simulatenous requests to download a File - could one get overwritten by the other with this naming "document.docx" approach?

Reply:
if temp files are given unique names, they will need to be cleaned up - see https://stackoverflow.com/questions/16691437/when-are-java-temporary-files-deleted